Jenkinsfile

```diff
@@ -0,0 +1,208 @@
+pipeline {
+
+    agent { label "coverage-worker" }
+
+    options {
+        timestamps()
+        timeout(75)
+    }
+
+    stages {
+        stage('Run Tests') {
+            parallel {
+                stage('lms_unit_1') {
+                    agent { label "jenkins-worker" }
+                    environment {
+                        SHARD = 1
+                        TEST_SUITE = 'lms-unit'
+                    }
+                    steps {
```

Comment:
@jmbowman It seems like there's a few approaches we can take with this in terms of trying to be more DRY. The way I currently have it extends much of the shared steps from each shard into a simple function that we can call by using a `script` block, which basically gives **declarative** pipelines access to **scripted** pipeline syntax.

As you probably saw from my numerous commits (prior to my squash), I had attempted to go even further, perhaps using some scripted syntax to generate loops for shards so that we truly aren't repeating any code. But it got to the point where the added confusion didn't seem worth it (at least in my opinion).

Also, in doing this research I stumbled upon **shared libraries** (more here: https://jenkins.io/doc/book/pipeline/shared-libraries/ ) which might be a better way of sorting out this reusable groovy in the future...  but it feels like we should separate that work from this (if we choose that path).

Reply:
So if I'm reading the shared libraries docs correctly - to use that approach to abstract out one step from stages running in parallel, we'd need a separate git repo with a minimum of 4 folders and 3 files, and switch that step from declarative to scripted syntax.  (Unless you want to abstract the entire pipeline, in which case it can stay declarative).  Ugh.  What's here currently is definitely an improvement.